src/njoy21/input/RECONR.hpp

```diff
@@ -68,7 +68,14 @@ class RECONR {
                                    std::move(card6) );
 
         try{
+          int previousMat = card3.mat.value;
           card3 = Card3( is );
+          if( card3.mat.value != 0 and previousMat > card3.mat.value ){
```

Comment:
we can remove the `while` at the end of the loop and just use a break statement to avoid redundant checks.